### DISCUSSION TITLE
Anyio version is out of current version: latest is 3.7.2, fastapi covers to 4.0.0

### DISCUSSION THREAD

**Discussion by a-yangyi** [fastapi/fastapi#9639] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# code in fastapi-0.86.0.dist-info/METADATA on my pc
Requires-Dist: anyio[trio]<4.0.0,>=3.2.1; extra == 'test'
```

### Description

In ```fastapi-0.86.0.dist-info/METADATA,  it defines anyio version as <3.2.1, 4.0.0>. (Didn't change much now https://github.com/tiangolo/fastapi/blob/155fc5e24e4bfde690dd2314c02d93b92ca5d78b/pyproject.toml#L72)
But the lastest version of anyio is 3.7.0 now.
Means it pulls the lastest version of anyio when I build my mirror, which may includes some bugs.

Recently I got an ''ASGI callable returned without starting response." error in my program, in the end I found it caused by anyio version. After I downgrade it to 3.6.2, the version before 3.7.0, the error disappeared.

Maybe we should define a known range to packages we use.

### Operating System

Linux

### FastAPI Version

fastapi-0.86.0

### Python Version

3.11.0

### Additional Context

Maybe related to:
https://github.com/tiangolo/fastapi/discussions/6644?sort=old

**Comment by Kludex** [comment, marked as answer]
A good article for this: https://hynek.me/articles/semver-will-not-save-you/

Also, you are checking the `test` extras, which is supposed to be used only for FastAPI test suite.
